### PR TITLE
fixing iteration information printed out for optimization.  

### LINKED ISSUE #17: clean up iteration info output and fix
The iteration info should include additional output for the total number of FEM solves.  It currently only tracks the number of gradient, objective, and hessian solves but each Hessian solve actually contains two FEM solves.  

It is also showing 1 objective and 1 gradient solve for iteration 0, should these also be 0 or has there been a solve on iteration 0.  The hessian iteration count is 2 for a force inversion problem and it should only be 1.    

This can all be checked by comparing the tao output written to the screen with the iteration info being written by the reporter.

### PR DESCRIPTION
**Pull request by lynnmunday**
It now counts the number of total FE solves 
closes #17

### CODE REVIEW

**Comment by lynnmunday** [comment]
@zachmprince will you review.  This was an issue raised during Murthy's Hessian PR.  The total number of FE solves is now tracked with a separate variable.

File: test/tests/optimizationreporter/objective_gradient_minimize/bc_load_linearFunction/homogeneous_forward.i [hunk 1]
```diff
@@ -128,7 +128,7 @@
 []
 
 [Outputs]
-  console = false
+  #console = false
```
Comment (zachmprince) [inline test/tests/optimizationreporter/objective_gradient_minimize/bc_load_linearFunction/homogeneous_forward.i:131]: Remove this line if you're going to comment it out.

File: test/tests/optimizationreporter/objective_gradient_minimize/bc_load_linearFunction/master.i [hunk 2]
```diff
@@ -86,8 +86,8 @@
     [fromForward]
       type = MultiAppReporterTransfer
       from_multi_app = forward
-      from_reporters = 'data_pt/temperature data_pt/temperature'
-      to_reporters = 'OptimizationReporter/simulation_values receiver/measured'
+      from_reporters = 'data_pt/temperature'
+      to_reporters = 'OptimizationReporter/simulation_values'
```
Comment (zachmprince) [inline test/tests/optimizationreporter/objective_gradient_minimize/bc_load_linearFunction/master.i:90]: The changes to these input seem unrelated to the purpose of this PR. What is the purpose?

**Review by zachmprince — CHANGES_REQUESTED** [review]

File: test/tests/optimizationreporter/objective_gradient_minimize/bc_load_linearFunction/master.i (hunk 2 above)
Comment (lynnmunday) [reply test/tests/optimizationreporter/objective_gradient_minimize/bc_load_linearFunction/master.i:90]: It isn't related to this PR.  I'm trying to clean up the input files so that transfers no longer write their data to a constant reporter on the main app.  All of the information written to the constant reporter is available in the OptimizationReporter so now I'm changing the csvdiffs to use that file and getting rid of the constant reporter.

Comment (lynnmunday) [reply test/tests/optimizationreporter/objective_gradient_minimize/bc_load_linearFunction/master.i:90]: I also changed the executioners to use Newton and line_search=none

**Comment by lynnmunday** [comment]
@zachmprince will review again

**Comment by zachmprince** [comment]
I'll merge this. But let's create an issue for cleaning up inputs. In the future, we should separate commits based on the issue they are resolving.

**Review by zachmprince — APPROVED** [review]
